tests/pages/firefox/all.py

```diff
@@ -0,0 +1,42 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from selenium.webdriver.common.by import By
+
+from pages.firefox.base import FirefoxBasePage, FirefoxBasePageRegion
+
+
+class FirefoxAllPage(FirefoxBasePage):
+
+    _url = '{base_url}/{locale}/firefox/all'
+
+    _search_input_locator = (By.ID, 'language-search-q')
+    _submit_button_locator = (By.CSS_SELECTOR, '#language-search button')
+    _build_rows_locator = (By.CSS_SELECTOR, '#builds table > tbody > tr')
+
+    @property
+    def builds(self):
+        return [Build(self, root=el) for el in
+                self.find_elements(self._build_rows_locator)]
+
+    @property
+    def displayed_builds(self):
+        return [Build(self, root=el) for el in
+                self.find_elements(self._build_rows_locator) if el.is_displayed()]
+
+    def builds_filtered_by_language(self, value):
+        languages = [el for el in self.builds if value in el.language]
+        return len(languages) == len(self.displayed_builds)
+
+    def search_for(self, value):
+        self.find_element(self._search_input_locator).send_keys(value)
+        self.find_element(self._submit_button_locator).click()
+        self.wait.until(lambda s: self.builds_filtered_by_language(value))
```

Comment:
This is essentially doing our assert before the test gets a chance to do it. Is there any other way to indicate that the search is complete? Have you seen that this wait is necessary?

If there's no other way and this is needed, then I would avoid the additional method that's only used here by moving the condition inline.

```python
expected_builds = [b for b in self._builds if value in b.language.lower()]
self.wait.until(lambda s: len(expected_builds) == len(self.displayed_builds))
```